estimagic/optimization/optimize.py

```diff
@@ -344,51 +466,67 @@ def _internal_minimize(
 
     bounds = tuple(params.query("_internal_free")[["lower", "upper"]].to_numpy().T)
 
-    try:
-        if origin in ["nlopt", "pygmo"]:
-            results = minimize_pygmo_np(
-                internal_criterion,
-                internal_params,
-                bounds,
-                origin,
-                algo_name,
-                algo_options,
-            )
-
-        elif origin == "scipy":
-            results = minimize_scipy_np(
-                internal_criterion, internal_params, bounds, algo_name, algo_options
-            )
-        elif origin == "tao":
-            results = minimize_pounders_np(
-                internal_criterion, internal_params, bounds, **algo_options
-            )
-        else:
-            raise NotImplementedError("Invalid algorithm requested.")
-
-    except Exception:
-        results = {
-            "optimization_status": "failure",
-            "traceback": traceback.format_exc(),
-        }
-        params = None
+    if database:
+        update_scalar_field(database, "optimization_status", "running")
+
+    if origin in ["nlopt", "pygmo"]:
+        results = minimize_pygmo_np(
+            internal_criterion,
+            internal_params,
+            bounds,
+            origin,
+            algo_name,
+            algo_options,
+            internal_gradient,
+        )
 
-    else:
-        results["optimization_status"] = "success"
-        params = reparametrize_from_internal(
-            internal=results["x"],
-            fixed_values=params["_internal_fixed_value"].to_numpy(),
-            pre_replacements=params["_pre_replacements"].to_numpy().astype(int),
-            processed_constraints=constraints,
-            post_replacements=params["_post_replacements"].to_numpy().astype(int),
-            processed_params=params,
+    elif origin == "scipy":
+        results = minimize_scipy_np(
+            internal_criterion,
+            internal_params,
+            bounds=bounds,
+            algo_name=algo_name,
+            algo_options=algo_options,
+            gradient=internal_gradient,
         )
+    elif origin == "tao":
+        crit_val = general_options["criterion"]
```

Comment:
I think it would be better to call this "start_criterion_value" to be very explicit. Just `criterion` usually refers to the function in estimagic, not to a specific function value.